scss/core/base.scss

```diff
@@ -352,7 +352,8 @@ table.mce-item-table {
 
 // Outline
 
-.focus-outline:focus:not(:hover) {
+.focus-outline:focus:not(:hover),
+.focus-within-outline:focus-within:not(:hover) {
```

Comment:
Can this not work? Why does it need a new class?

```suggestion
.focus-outline:focus-within:not(:hover) {
```

Reply:
So we have 2 classes we can use.
One class is to focus an element directly.
Another class if for situations were for example input is located inside a label. In that case, we give label second class which works on focus:within.

Reply:
For example, star rating component will work like this.

![focus-within-outline](https://user-images.githubusercontent.com/52824207/88954871-b536ef80-d2a3-11ea-86f9-8c5a48caf3b1.PNG)

Reply:
I understand what you're trying to do, but I don't understand why we can't use the `focus-outline` for both scenarios.

```html
<div class="focus-outline">
  <input tabindex="0" />
</div>

<span class="btn btn-primary class="focus-outline" tabindex="0">Button</span>
```

```css
.focus-outline:focus:not(:hover),
.focus-outline:focus-within:not(:hover) {
  /* box-shadow: ... */
}
```

This would equally work, no? `.focus-outline` determines which element the outline will be applied. The focus can be on the element, or an element within.

Reply:
I understood now. Yes, I  think that would work.

Reply:
Fixed.